kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/util.kt

```diff
@@ -448,3 +448,21 @@ internal fun KtValueParameterSymbol.getDefaultValue(): KtAnnotationValue? {
         }
     }
 }
+
+fun interface Restorable {
+    fun restore(): KSAnnotated?
+}
+
+fun interface Deferrable {
+    fun defer(): Restorable?
+}
+
+fun <T : KtSymbol> T.defer(restore: (T) -> KSAnnotated?): Restorable? {
+    val ptr = analyze { with(this) { this@defer.createPointer() } }
```

Comment:
IIRC back to early days I had some discussion with JB regarding the usability of symbol pointers and my impression was that this might not suite best for our case, have you chatted with them on using pointers?

Reply:
Would you mind elaborating?